x-pack/plugins/apm/public/components/shared/Links/APMLink.tsx

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiLink, EuiLinkAnchorProps } from '@elastic/eui';
+import React from 'react';
+import url from 'url';
+import { useLocation } from '../../../hooks/useLocation';
+import { APMQueryParams, getSearchWithCurrentTimeRange } from './url_helpers';
+
+interface Props extends EuiLinkAnchorProps {
+  path?: string;
+  query?: APMQueryParams;
+}
+
+const APMLink: React.FC<Props> = ({ path, query, ...rest }) => {
```

Comment:
I've noticed yet another problem with this approach. We've already talked about how this notation make it hard to statically infer, and thus removing the component name, which affects tests and Dev Tools.

It also impedes vscode's (and perhaps other IDEs) ability to "go to defintion". When I click on an instance of  `APMLink` I get a suggestion for both `FunctionComponent` and `APMLink`. This is fairly annoying when navigating around:

![Screenshot 2019-04-08 at 14 33 53](https://user-images.githubusercontent.com/209966/55724275-5f359d00-5a0b-11e9-94d4-a515fdebab36.png)

Reply:
I'm still not seeing all the benefits that would offset the disadvantages - what am I missing?

Reply:
> We've already talked about how this notation makes static analysis harder, and thus removes the component name, which affects tests and Dev Tools.

I think this only happens if you do `export const APMLink =`, which is why the `export` has been moved to a different line. The two definitions are expected since one is basically extending the other.

Since React provides a specific type for its function components, it seems like a good idea to use it if possible, which is why I've tried to use it when I can. I especially don't like having to manually add `children` to our own Props interface and then guess if it should be `React.Node` or one of the other node-like types there, etc., rather than just allowing React to handle that. 

I wish babel would fix this tbh, and just allow us to do `export const APMLink =` without losing the function name.

Reply:
> The two definitions are expected since one is basically extending the other.

Sure, vscode is doing the right thing - but as a dev when I click on go-to-def for `APMLink` I just want to go the place where `APMLink` is defined. That's what happens when you `export function() {...}